src/peep/peep.c

```diff
@@ -6160,14 +6171,15 @@ void peep_set_map_tooltip(rct_peep *peep)
 
 
 void sub_693BAB(rct_peep* peep) {
-	uint8 bl = peep->var_6F;
-	if (bl != peep->action_sprite_type) {
+	// Add spriteType as function parameter and make peep->next_action_sprite_type obsolete?
+	uint8 spriteType = peep->next_action_sprite_type;
+	if (spriteType != peep->action_sprite_type) {
 		invalidate_sprite_2((rct_sprite*)peep);
-		peep->action_sprite_type = bl;
+		peep->action_sprite_type = spriteType;
 		uint8* edx = RCT2_ADDRESS(0x98270C, uint8*)[peep->sprite_type * 2];
```

Comment:
I'm confused. To me, it looks like at 0x00982708 starts an array of (pairs of pointers) which each point to arrays of the struct you describe (for example starting at 0x9830a0). This line accesses the second pointer of the pair.